addons/utm/models/utm_mixin.py

```diff
@@ -20,13 +20,19 @@ class UtmMixin(models.AbstractModel):
     @api.model
     def default_get(self, fields):
         values = super(UtmMixin, self).default_get(fields)
-
         # We ignore UTM for salemen, except some requests that could be done as superuser_id to bypass access rights.
         if self.env.uid != SUPERUSER_ID and self.env.user.has_group('sales_team.group_sale_salesman'):
             return values
 
+        values.update(self._set_utm_values(values))
+        return values
+
+    def _set_utm_values(self, values_list, allow_create=False):
```

Comment:
Maybe we could rename to _get_missing_utm_values, as we now have a values_list (to rename to values as it is normally not a list :) ) that is used as base to complete. That way we know we don't set them, just receive the missing ones.